#423/chat redesign

camelPhonso opened the pull request:
# Description

**Closes #423**

Restyles the `MessageCard` and `MessageForm` components according to the new figma design. 

Some refactoring is done but all in line with the scope of matching the new design. I see some of the e2e tests are broken on `dev` but to fix them seemed to be too large of a scope creep for this. I did add some adjustments to the general display of the conversation because it just made sense (for example, MessageCards were extending beyond the window on dev apparently)

### Files changed

- `MessageCard.tsx`, `messaging-styles.css`, `PaperPlaneIcon.tsx`,  `MessageForm.tsx`, `MessageForm.stories.tsx` - re-styling to match new figma
- `ReadReceiptIcon.tsx` && `CurrentConversation.tsx` - created new icon component and adjusted the props being passed to the `MessageCard`s to put in place the read-receipt feature that's now part of the design.

### UI changes

All as described above. Looks like this:

<img width="1109" alt="Screenshot 2025-06-22 at 22 19 29" src="https://github.com/user-attachments/assets/e9898399-d0a4-4532-95c3-a3bdec72cd45" />

<img width="397" alt="Screenshot 2025-06-22 at 22 21 05" src="https://github.com/user-attachments/assets/0f50cf04-14dd-4926-92d5-8ff277b19120" />


### Changes to Documentation

Added the `MessageForm` component to our component library:

<img width="1665" alt="Screenshot 2025-06-22 at 22 23 09" src="https://github.com/user-attachments/assets/5d3cbf34-8e59-4825-839b-519f0510c3ef" />


# Tests

Only one test in the messaging suite is currently broken. I did try to debug the issue but it seems like it's worth it's own task to fix as it's something around dispatching our 'delete_conversation' action in the store.

camelPhonso committed:
style: add user name to messagecard display

relates #423

camelPhonso committed:
style: round corners dynamically with css logical properties

relates #423

camelPhonso committed:
style: correct name and time placement according to logical direction property
relates #423

camelPhonso committed:
style: set new colours for messaging card according to sender

relates #423

camelPhonso committed:
feat: create new dynamic icon for use in read-receipt feature

relates #423

camelPhonso committed:
style: display read-receipt icon for messages from self

relates #423

camelPhonso committed:
refactor: remove internal state to use existing data property for read status instead

relates #423

camelPhonso committed:
feat: pass is_read value to message card component for read receipt display

relates #423

camelPhonso committed:
docs: add MessagingForm stories to our component library

relates #423

camelPhonso committed:
style: adjust colour scheme to new design

relates #423

camelPhonso committed:
style: set new svg based on updated design

relates #423

camelPhonso committed:
style: remove rounded border from submit button

relates #423

camelPhonso committed:
style: place submit button inside visible text area

relates #423

camelPhonso committed:
style: set bottom margin

relates #423

camelPhonso committed:
style: keep user name on left-hand of card display in small screens

relates #423

camelPhonso committed:
merge dev

nichgalzin commented on components/messaging/MessageCard.tsx:
```diff
@@ -30,18 +33,25 @@ const MessageCard: React.FC<MessageCardProps> = ({
+        className={`${isCurrentUser ? 'flex-row lg:mr-24' : 'flex-row-reverse text-end'} align-center flex w-full justify-between`}
+      >
+        <p className='text-sm font-bold'>{`${isCurrentUser ? 'You' : currentUser}`}</p>
+        <div className='align-center flex w-fit justify-between'>
```
I'm seeing the other users Id rather then their user name here. 

<img width="680" alt="Screenshot 2025-06-27 at 11 25 00" src="https://github.com/user-attachments/assets/f6d5a6fd-b6bc-4fa1-96c8-02a1be154be8" />

camelPhonso replied:
Fixed this in a new commit 👍

nichgalzin reviewed (changes requested):
Hello darling! Thanks for this! Nice and tidy. Just a couple of notes.

- Longer messages are covering the send icon and the margin to the bottom of the screen is too cramped when the message extends to more then one line. 
<img width="378" alt="Screenshot 2025-06-27 at 10 33 34" src="https://github.com/user-attachments/assets/eb6992f8-09f6-46a9-a335-d8bbf7010e60" />
- Also if you like I'm organising components with there stroybook file into sub-directories as I work on them, so as do to them incrementally like I did with with conversation card etc. You could do that if you like WDUT?

camelPhonso committed:
style: add padding to prevent text from overtaking the paper plane icon

relates #423

camelPhonso committed:
style: set hight as hight-min value to allow the text area to resize for longer text

relates #423

camelPhonso committed:
feat: retrieve user name for conversation partner

relates #423

camelPhonso committed:
build: grouped components with story files in dedicated folder

relates #423

camelPhonso commented:
Hey-o, thanks! Completely missed those. I've made a few changes and it should all be good now, see here:


> * Longer messages are covering the send icon and the margin to the bottom of the screen is too cramped when the message extends to more then one line.

Added some padding to fix this. Working better now on fullscreen and mobile. 

> - Also if you like I'm organising components with there stroybook file into sub-directories as I work on them, so as do to them incrementally like I did with with conversation card etc. You could do that if you like WDUT?

I've moved the components I touch to a dedicated folder. I do like this way for organising it but keep in mind that merging this change will mess with the git history of those files, which might just make it weird to handle some merge conflicts in upcoming merges.

This is how it's looking:
<img width="326" alt="Screenshot 2025-06-29 at 15 26 32" src="https://github.com/user-attachments/assets/bac48a25-d759-465b-8ee6-d294bd9b6838" />

nichgalzin reviewed (approved):
Amazing babes! So happy to have you contributing again.
